app/controllers/events_controller.rb

```diff
@@ -133,6 +120,19 @@ def destroy
 
   private
 
+  def add_collective_members_and_trainees
+    if params["trainees"]
+      params["trainees"].each do |t|
+        @event.trainees << Trainee.find(t.to_i)
+      end
+    end
+    if params["collectiveMembers"]
+      params["collectiveMembers"].each do |m|
+        @event.collective_members << CollectiveMember.find(m.to_i)
+      end
+    end
+  end
+
```

Comment:
setting collective members and trainees on an event is now a private method that's always called and is event-agnostic